system/lib/libc/musl/src/misc/emscripten_pthread.c

```diff
@@ -10,13 +10,16 @@ pthread_t pthread_self(void) {
 #endif // !__EMSCRIPTEN_PTHREADS__
 
 #if __EMSCRIPTEN_PTHREADS__
-// Needs to be called after PThread.initRuntime, which in turn needs to be
-// called after constructors have run and memory is initialized.
+EM_JS(void, initPthreadsJS, (void), {
+  PThread.initRuntime();
+})
+#endif
+
 EMSCRIPTEN_KEEPALIVE
```

Comment:
This EMSCRIPTEN_KEEPALIVE used to be conditional but now it is not. Is that intentional?

Reply:
Not intentional, yeah, and I see it broke a metadce test, good we kept those around ;)